source/common/stats/thread_local_store.h

```diff
@@ -225,6 +225,20 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
     Histogram& histogram(const std::string& name) override;
     Histogram& tlsHistogram(const std::string& name, ParentHistogramImpl& parent) override;
 
+    template <class StatType>
+    using MakeStatFn = StatType* (StatDataAllocator::*)(const std::string& name,
+                                                        std::string& tag_extracted_name,
+                                                        std::vector<Tag>& tags);
+
+    // Makes a stat either by looking it up in the central cache,
+    // generating it from the the parent allocator, or as a last
+    // result, creating it with the heap allocator.
+    template <class StatType>
+    StatType&
+    safeMakeStat(const std::string& name,
```

Comment:
Can we add more documentation here about each argument.  I'm only asking because the signature is reasonably complicated.

Reply:
OK I left some.  I'm not sure it's super-illuminating as the semantics around tls_are rather complex :)